src/extensions/default/RelatedFiles/main.js

```diff
@@ -0,0 +1,202 @@
+            var document = new DOMParser().parseFromString(docText, "text/html");
+        
+            var linkTags = $(document).find("link[rel='stylesheet']");
+            var scriptTags = $(document).find("script[src!=null]");
+            
+        	var i, fileURI, searchResult;
+			
+            // Search for href in link tag to get related file.
+            for (i = 0; i < linkTags.length; i++) {
+                var linkHref = linkTags[i].getAttribute("href");
+                if (linkHref !== null) {
+                    if (isRemoteURL(linkHref)) {
+                        searchResult = getLinkFromRemoteURL(linkHref);
+                    }
+                    else if (PathUtils.isAbsoluteUrl(linkHref)) {
+                        searchResult = getLinkFromLocalURL(linkHref);
+                    }
+                    else {
+                        fileURI = getAbsoultePath(currentDocument.file.fullPath, linkHref);
+                        searchResult = getLinkFromLocalURL(fileURI);
+                    }
+                    if (searchResult) {
+                        relatedFiles.push(searchResult);    
+                    }
+                }
+            }    
+            // Search for src in script tag to get related files.
+            for (i = 0; i < scriptTags.length; i++) {
+                var scriptSrc = scriptTags[i].getAttribute("src");
+                if (scriptSrc !== null) {
+                    if (isRemoteURL(scriptSrc)) {
+                        searchResult = getLinkFromRemoteURL(scriptSrc);
+                    }
+                    else if (PathUtils.isAbsoluteUrl(scriptSrc)) {
+                        searchResult = getLinkFromLocalURL(scriptSrc);
+                    }
+                    else {
+                        fileURI = getAbsoultePath(currentDocument.file.fullPath, scriptSrc);
+                        searchResult = getLinkFromLocalURL(fileURI);
+                    }
+                    if (searchResult) {
+                        relatedFiles.push(searchResult);    
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+    * Check whether a given string is remote url.
+    * @param {string} tagurl The url which is to be checked for remote link.
+    * @returns {Boolean} true if the url matches regular expression for remote url.
+    **/
+	function isRemoteURL(tagurl) {
+		var pattern = RegExp('^((http|https|ftp)?:?\/\/)');
+        return pattern.test(tagurl); 
+	} 
+    
+    /**
+    * Create a SearchResult object which has path information about the local file.
+    * @params {string} tagurl The url from which file path information is to be extracted.
+    * @returns {Object} SearchResult object which has file path information.
+    **/
+    function getLinkFromLocalURL(tagurl) {
+        var searchResult = new StringMatch.SearchResult (tagurl);
+        var file = Filesystem.getFileForPath(tagurl);
+        searchResult.label = file._name;
```

Comment:
`searchResult` entry creation can be extracted as a parameterised function and called from `getLinkFromLocalURL` and `getLinkFromRemoteURL`.